compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -4923,6 +4943,321 @@ private Map<String, Location> getFieldLocations(List<BLangSimpleVariable> fields
+        PackageID packageID = bLangPackage.packageID;
+
+        List<BLangVariable> moduleVarsAndConstants = new ArrayList<>() {{
+            addAll(bLangPackage.globalVars);
+            addAll(bLangPackage.constants);
+        }};
+
+        for (BLangVariable construct : moduleVarsAndConstants) {
+            if (!construct.flagSet.contains(Flag.PUBLIC)) {
+                continue;
+            }
+
+            BLangType typeNode = construct.typeNode;
+
+            if (typeNode == null) {
+                continue;
+            }
+
+            logErrorForUnsupportedConstructExposureFromClientDeclModule(packageID, collector, construct.pos,
+                                                                        typeNode.getBType());
+        }
+
+        for (BLangTypeDefinition typeDefinition : bLangPackage.typeDefinitions) {
+            Set<Flag> flagSet = typeDefinition.flagSet;
+            if (!flagSet.contains(Flag.PUBLIC) || flagSet.contains(Flag.ANONYMOUS)) {
+                continue;
+            }
+
+            logErrorForUnsupportedConstructExposureFromClientDeclModule(packageID, collector, typeDefinition.pos,
+                                                                        typeDefinition.typeNode.getBType());
+        }
+
+        for (BLangClassDefinition classDefinition : bLangPackage.classDefinitions) {
+            Set<Flag> flagSet = classDefinition.flagSet;
+            if (!flagSet.contains(Flag.PUBLIC) || classDefinition.isServiceDecl || flagSet.contains(Flag.OBJECT_CTOR)) {
+                continue;
+            }
+
+            logErrorForUnsupportedConstructExposureFromClientDeclModule(packageID, collector, classDefinition.pos,
+                                                                        classDefinition.getBType());
+        }
+
+        for (BLangFunction function : bLangPackage.functions) {
+            if (!function.flagSet.contains(Flag.PUBLIC) &&
+                    (!function.attachedFunction || !function.receiver.flagSet.contains(Flag.PUBLIC))) {
+                continue;
+            }
+
+            logErrorForUnsupportedConstructExposureFromClientDeclModule(packageID, collector, function.pos,
+                                                                        function.getBType());
+        }
+    }
+
+    private void logErrorForUnsupportedConstructExposureFromClientDeclModule(PackageID currentPkgId,
+                         BClientDeclConstructExposureDisallowingTypeVisitor collector, Location location, BType type) {
+        List<BType> exposedTypes = new ArrayList<>(1);
+        collector.visitType(type, exposedTypes, currentPkgId);
+        if (!exposedTypes.isEmpty()) {
+            dlog.error(location,
+                       DiagnosticErrorCode.UNSUPPORTED_EXPOSURE_OF_CONSTRUCT_FROM_MODULE_GENERATED_FOR_CLIENT_DECL);
+        }
+    }
+
+    private static class BClientDeclConstructExposureDisallowingTypeVisitor implements TypeVisitor {
+
```

Comment:
We can remove this line

Reply:
Any specific reason? I prefer having it as it since it's clearer at first glance. We've done the same even for the `SemanticAnalyzer` class.

Reply:
I suggest it because, the other methods in this class haven't new line at the top.

Reply:
But this is a class?

Reply:
yes. Got it now. Then agree with you